src/ejabberd_c2s.erl

```diff
@@ -2800,17 +2805,26 @@ stream_mgmt_handle_ack(NextState, El, #state{} = SD) ->
             maybe_send_element_from_server_jid_safe(SD, mongoose_xmpp_errors:invalid_namespace()),
             maybe_send_trailer_safe(SD),
             {stop, normal, SD};
-        throw:{policy_violation, Reason} ->
-            PolicyViolation = mongoose_xmpp_errors:policy_violation(SD#state.lang, Reason),
-            maybe_send_element_from_server_jid_safe(SD, PolicyViolation),
+        error:badarg ->
```

Comment:
isn't it too wide matching?
Maybe introducing parse_h, that throws some error we can match would be nicer.